socorro/external/boto/connection_context.py

```diff
@@ -399,3 +379,44 @@ def _connect(self):
                 **self._get_credentials()
             )
             return self.connection
+
+
+class HostPortS3ConnectionContext(S3ConnectionContext):
+    """Connection context for connecting to an S3-like service at a specified
+    host/port
+
+    This is useful if you're connecting to a fake s3 or minio or some other
+    non-S3 thing.
+
+    """
+
+    required_config = Namespace()
+    required_config.add_option(
+        'host',
+        doc='The hostname to connect to',
+        reference_value_from='resource.boto',
+    )
+    required_config.add_option(
+        'port',
+        doc='The network port',
+        reference_value_from='resource.boto',
+        from_string_converter=int
+    )
+    required_config.add_option(
+        'secure',
+        doc='Whether to connect securely or not (true/false)',
+        reference_value_from='resource.boto',
+        from_string_converter=lambda x: x.lower().startswith('t'),
```

Comment:
configman says it has a bool parser, but its bool parser is the built-in Python type which means an empty string is False and any non-empty string is True. I think that kind of sucks for configuration, so I wrote this which results in more readable configuration values True and False.

What we should probably do is what other configuration libraries do and implement a bool parser that handles a variety of different positive and negative values in configman. I wrote up an issue along those lines:

https://github.com/mozilla/configman/issues/165

Until then, I'm just going to leave this here.